pkg/controller/etcdproxy/certs.go

```diff
@@ -86,27 +84,39 @@ func (c *EtcdProxyController) setNewAPIServerCertificates(etcdstorage *etcdstora
 		return err
 	}
 
-	// Generate client certificate/key pair.
-	clientBundle, err := clientSigner.NewClientCertificate(pkix.Name{CommonName: "client"},
-		r.Int63n(100000), currentTime)
-	if err != nil {
-		return err
-	}
-	clientCertBytes, err := certs.EncodeCertificates(clientBundle.Certificates...) // goes to apiserver
-	if err != nil {
-		return err
-	}
-	clientKeyBytes, err := certs.EncodeKey(clientBundle.Key) // goes to apiserver
+	// Write CA certificate to EtcdProxyController CA ConfigMap.
+	err = c.createEtcdProxyClientCAConfigMap(etcdstorage, clientSignerCert)
 	if err != nil {
 		return err
 	}
 
-	// Write pairs to appropriate ConfigMaps and Secrets.
-	err = c.createEtcdProxyClientCAConfigMap(etcdstorage, clientSignerCert)
-	if err != nil {
-		return err
+	// Generate client certificate for each Secret provided.
+	var errs []error
+	for _, secret := range etcdstorage.Spec.ClientCertSecrets {
+		// Generate client certificate/key pair.
+		clientBundle, err := clientSigner.NewClientCertificate(pkix.Name{CommonName: "client"},
```

Comment:
When you create separate client certs, you'll want to have more unique names.  You have a secret namespace and name.  How about include those in here?